packages/core/upload/admin/src/utils/getFolderURL.js

```diff
@@ -0,0 +1,18 @@
+import { stringify } from 'qs';
+
+const getFolderURL = (pathname, query, folder) => {
+  const { _q, ...queryParamsWithoutQ } = query;
+  const queryParamsString = stringify(
+    {
+      ...queryParamsWithoutQ,
+      folder: folder?.id,
+    },
+    { encode: false }
+  );
+
+  // Search query will always fetch the same results
+  // we remove it here to allow navigating in a folder and see the result of this navigation
+  return `${pathname}${queryParamsString ? `?${queryParamsString}` : ''}`;
```

Comment:
I think you don't have to change it, but it would be much nicer to use the [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL) for such as case.

Reply:
interesting, since we use this stringify format at multiple places within the app I would propose to add this to my backlog for another PR and fix it everywhere instead of just here, wdyt?